src/mainClass.js

```diff
@@ -2,44 +2,76 @@ import { getData, getAllCategories } from "./source";
 
 export class AllItems {
 
-  constructor(searchCategory, searchLinks, displayLinksForChosenCategory, searchChosenLinksOnly) {
+  constructor(searchCategory, searchLinks, displayLinksForChosenCategory) {
     this.categories = [];
     this.onReceivedCategories = searchCategory;
     this.displayChosenLinks = displayLinksForChosenCategory;
     this.links = [];
     this.onReceivedLinks = searchLinks;
-    this.chosenLinks = [];
-    this.onReceivedChosenLinks = searchChosenLinksOnly;
+    this.storageLinks = JSON.parse(localStorage.getItem('links'));
+    this.storageCategories = JSON.parse(localStorage.getItem('categories'));
+    this.filtered;
   }
 
-  setOnCategoriesReceived(onReceivedCategories) { 
-    this.onReceivedCategories = onReceivedCategories 
+  setOnCategoriesReceived(onReceivedCategories) {
+    this.onReceivedCategories = onReceivedCategories
   }
 
   setOnLinksReceived(onReceivedLinks) {
     this.onReceivedLinks = onReceivedLinks;
   }
 
-  setOnChosenLinksReceived(onReceivedChosenLinks) {
-    this.onReceivedChosenLinks = onReceivedChosenLinks;
-  }
-
   setDisplayChosenLinks(displayChosenLinks) {
     this.displayChosenLinks = displayChosenLinks;
   }
 
+  updateStorageLinks() {
+    localStorage.setItem('links', JSON.stringify(this.links));
+  }
+
+  updateStorageCategories() {
+    localStorage.setItem('categories', JSON.stringify(this.categories));
+  }
+
+  markAsImportant(el) {
+    if (el.important === false) {
+      el.important = true;
+    } else {
+      el.important = false;
+    }
+  }
+
+  setFiltered(param) {
+    this.filtered = param;
+  }
+
+  getAllLinks(filtered, e) {
+    if (filtered !== null) {
+      let filteredCategoriesAfterSearch = this.links.filter(el => {
+        if (el.API.toLowerCase().includes(e.target.value.toLowerCase())) {
```

Comment:
what is this: el.API.toLowerCase()?
And this is wrong: e.target.value.toLowerCase(), not send event, send value to method from ui, not whole event.

Reply:
el.API is a name of the link.

eg. :
API: "Cat Facts"
Auth: ""
Category: "Animals"
Cors: "no"
Description: "Daily cat facts"
HTTPS: true
Link: "https://alexwohlbruck.github.io/cat-facts/"
important: false

so el.API.toLowerCase() will be "cat facts";

What am i doing here is comparsion of an input value in search bar with el.API (name of an link), so if el.API includes letters that i am typing into search bar, it will display those elements;